packages/builder/src/components/design/settings/controls/FieldConfiguration/FieldConfiguration.svelte

```diff
@@ -38,20 +43,31 @@
 
   $: datasource = getDatasourceForProvider($currentAsset, componentInstance)
   $: resourceId = datasource?.resourceId || datasource?.tableId
+  let previousResourceId = null
 
   $: if (!isEqual(value, cachedValue)) {
     cachedValue = cloneDeep(value)
   }
 
   const updateState = value => {
     schema = getSchema($currentAsset, datasource)
+    primaryDisplayColumn = getPrimaryDisplayColumn($currentAsset, datasource)
     options = Object.keys(schema || {})
     sanitisedValue = getValidColumns(convertOldFieldFormat(value), options)
     updateSanitsedFields(sanitisedValue)
     unconfigured = buildUnconfiguredOptions(schema, sanitisedFields)
     fieldList = [...sanitisedFields, ...unconfigured]
       .map(buildPseudoInstance)
       .filter(x => x != null)
+
+    if (resourceId !== previousResourceId) {
```

Comment:
This autoselect all columns when changing datasources which is nicer imo.

Reply:
No longer making changes to this file, but leaving this in seems like a nice upgrade if no one objects 🤷

Reply:
I'm not sure that this behaviour is necessary as [this line](https://github.com/Budibase/budibase/blob/f0a162c8dc8c793deb72dea26f01e3014beade44/packages/builder/src/components/design/settings/controls/FieldConfiguration/FieldConfiguration.svelte#L83) will set the toggle to active when the resource is reset.

Reply:
Not 100% sure, want to say I tested it originally and it didn't seem to behave that way, but at this point I think it's best I just remove it regardless, so I have done so 🧙